src/service/extensions-impl.js

```diff
@@ -452,27 +454,29 @@ export class Extensions {
 /**
  * Calculate script url for amp-ad.
  * @visibleForTesting
+ * @param {string} protocol Serving protocol
  * @param {string} path Location path of the window
+ * @param {string} port Serving port
  * @param {string} extensionId
  * @param {boolean=} isTest
  * @param {boolean=} isUsingCompiledJs
  * @return {string}
  * @visibleForTesting
  */
-export function calculateExtensionScriptUrl(path, extensionId, isTest,
-    isUsingCompiledJs) {
+export function calculateExtensionScriptUrl(protocol, path, port, extensionId,
+    isTest, isUsingCompiledJs) {
   if (getMode().localDev) {
     if (isTest) {
       if (isUsingCompiledJs) {
         return `/base/dist/v0/${extensionId}-0.1.js`;
       }
       return `/base/dist/v0/${extensionId}-0.1.max.js`;
     }
-    if (path.indexOf('.max') >= 0) {
-      return `http://localhost:8000/dist/v0/${extensionId}-0.1.max.js`;
+    if (path.indexOf('.max') >= 0 || path.substr(0, 5) == '/max/') {
+      return `${protocol}//localhost:${port}/dist/v0/${extensionId}-0.1.max.js`;
```

Comment:
Confirmed that the `.min` path behavior differing from the `.max` path behavior predates this change by checking out `master`, doing a `gulp clean`, `gulp build`, `gulp dist`, and `gulp`.  In that context, loading `http://localhost:8000/examples/a4a.amp.max.html?a4a=2` causes `amp-ad-0.1.js` to be loaded from localhost and successfully branches to and loads `amp-ad-network-adsense-impl-0.1.js`, also from localhost.

However, when I load `http://localhost:8000/examples/a4a.amp.min.html?a4a=2`, `amp-ad-0.1.js` is loaded from `cdn.ampproject.org` instead and the a4a branch is never taken and `amp-ad-network-adsense-impl` is never loaded.  So there's an issue here, but it's not part of this PR.

If you all are otherwise happy with this PR, I suggest we pull this one and I'll open an issue on the `.min` loading behavior.  @lannka -- what do you think?